core/src/main/java/nl/nn/adapterframework/jdbc/migration/Migrator.java

```diff
@@ -66,8 +71,10 @@ public synchronized void configure(Configuration configuration, String changeLog
 			log.debug(msg);
 		}
 		else {
-			String dataSource = appConstants.getString("jdbc.migrator.dataSource", appConstants.getResolvedProperty("jdbc.datasource.default"));
-			setDatasourceName(dataSource);
+			if(StringUtils.isEmpty(getDatasourceName())) {
+				String dataSource = appConstants.getString("jdbc.migrator.dataSource", appConstants.getResolvedProperty("jdbc.datasource.default"));
+				setDatasourceName(dataSource);
+			}
```

Comment:
User will be able select the datasource from the GUI. Selected datasource will be used to generate the target database compatible SQL script. In case it is the ibiscontext initializing the migrator then the jdbc.migrator.dataSource will be used to determine the datasource